relayer/chains/cosmos/types.go

```diff
@@ -0,0 +1,42 @@
+package cosmos
+
+import (
+	clienttypes "github.com/cosmos/ibc-go/v3/modules/core/02-client/types"
+	chantypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+)
+
+// TransactionMessage is the type used for parsing IBC messages
+type TransactionMessage struct {
+	Action      string
+	PacketInfo  *PacketInfo
+	ChannelInfo *ChannelInfo
+	ClientInfo  *ClientInfo
+	// TODO ConnectionInfo
+}
+
+// ChannelInfo contains useful channel information during channel state changes
+type ChannelInfo struct {
+	ConnectionID             string
+	ClientID                 string
+	PortID                   string
+	ChannelID                string
+	CounterpartyClientID     string
+	CounterpartyConnectionID string
+	CounterpartyPortID       string
+	CounterpartyChannelID    string
+}
+
+// PacketInfo contains pertinent packet information for constructing IBC messages for the counterparty.
+type PacketInfo struct {
```

Comment:
Should we separate this into a Packet and Acknowledgement struct?

Reply:
similar to the above, I was thinking this could be the type for all possible packet properties parsed from the events. the ibc message handlers in a future PR can construct the more concrete types before anything leaves the scope of the `CosmosChainProcessor`.